minidump-common/src/traits.rs

```diff
@@ -19,7 +19,7 @@ pub trait Module {
     /// builds of the same code module.  This may contain a uuid, timestamp,
     /// version number, or any combination of this or other information, in an
     /// implementation-defined format.
-    fn code_identifier(&self) -> CodeId;
+    fn code_identifier(&self) -> Option<CodeId>;
```

Comment:
Would it be good to document on the trait why this can be `None`?  Describe somehow that some minidump writers write non-object things like mmapped fonts etc into the module stream and that for those this can not provide a valid CodeId.

(while adding that it could be referred to in `.debug_identifier()` as well)

Reply:
I think some documentation on this may be useful, but I have mixed feelings about adding this note to the trait itself, as these platform and/or implementation-specific details may not be applicable to every implementation of the trait (i.e. would this be applicable to `SimpleModule`?) 

It does look like the modules that fall into the `None` case are coming from shared memory segments (It's a bunch of stuff sitting in System V shared memory) which sounds quite similar to mmapped fonts, so I'll leave a comment above the branch noting this. Hopefully this works as a good compromise 🤞